docs/user/metal/install_upi.md

```diff
@@ -0,0 +1,305 @@
+
+## Getting ignition configs for machines
+
+The OpenShift Installer provides administrators various assets that are required to create an OpenShift cluster, namely:
+
+* Ignition configs: The OpenShift Installer provides Ignition configs that should be used to configure the RHCOS based boostrap and control plane machines using `bootstrap.ign`  and `master.ign` respectively. The OpenShift Installer also provides `worker.ign` that can be used to configure the RHCOS based `worker` machines, but also can be used as source for configuring RHEL based machines [todo-link-to-BYO-RHEL].
+
+* Admin Kubeconfig: The OpenShift Installer provides a kubeconfig with admin level privileges to Kubernetes APIServer.
+
+    NOTE: This kubeconfig is configured to use `api.$cluster_name.$base_domain` DNS name to communicate with the Kubernetes APIServer.
+
+### Setting up install-config for installer
+
+The OpenShift installer uses an [Install Config][install-config] to drive all install time configuration.
+
+An example install config for bare metal UPI is as follows:
+
+```yaml
+apiVersion: v1beta3
+## The base domain of the cluster. All DNS records will be sub-domains of this base and will also include the cluster name.
+baseDomain: example.com
+compute:
+- name: worker
+  replicas: 1
+controlPlane:
+  name: master
+  replicas: 1
+metadata:
+  ## The name for the cluster
+  name: test
+platform:
+  none: {}
+## The pull secret that provides components in the cluster access to images for OpenShift components.
+pullSecret: ''
+## The default SSH key that will be programmed for `core` user.
+sshKey: ''
+```
+
+Create a directory that will be used by the OpenShift installer to provide all the assets. For example `test-bare-metal`,
+
+```console
+$ mkdir test-bare-metal
+$ tree test-bare-metal
+test-bare-metal
+
+0 directories, 0 files
+```
+
+Copy *your* `install-config` to the `INSTALL_DIR`. For example using the `test-bare-metal` as our `INSTALL_DIR`,
+
+```console
+$ cp <your-instal-config> test-bare-metal/install-config.yaml
+$ tree test-bare-metal
+test-bare-metal
+└── install-config.yaml
+
+0 directories, 1 file
+```
+
+NOTE: The filename for `install-config` in the `INSTALL_DIR` must be `install-config.yaml`
+
+### Invoking the installer to get Ignition configs
+
+Given that you have setup the `INSTALL_DIR` with the appropriate `install-config`, you can create the Ignition configs by using the `create ignition-configs` target. For example,
+
+```console
+$ openshift-install --dir test-bare-metal create ignition-configs
+INFO Consuming "Install Config" from target directory
+$ tree test-bare-metal
+test-bare-metal
+├── auth
+│   └── kubeconfig
+├── bootstrap.ign
+├── master.ign
+└── worker.ign
+
+1 directory, 4 files
+```
+
+## Getting OS related assets for machines
+
+TODO RHEL CoreOS does not have assets for bare metal.
```

Comment:
Is this to include where to put the *.ign files which are referenced later in pxe config or should we go ahead and mention hosting those files in the previous section?

Reply:
this is how to get RHCOS iso / os-raw-image ..

Reply:
Ok, I think in the previous section we should mention the need to store *.ign files somewhere that's accessible to the nodes.

Reply:
https://github.com/openshift/installer/pull/1416/files/972d6a56d436fba77a4d7b70d653ee614b32472e#diff-1a64e7030769df4d25b1c09ddb8f62deR199 
`coreos-installer` is how RHCOS boots on metal. and it has `coreos.inst.ignition_url` kernel parameter